### PR TITLE
 Use enrollment_state when requesting courses

### PR DESCRIPTION
**Pull request by armstrongnate**
refs: MBL-13099
affects: student, teacher
release note: Only show active courses in Submit Assignment flow

The `active` enrollment state will include _unpublished_ courses for
teachers but will not include them for students which is what we want.

We still need to use `state` in RN call because we need past enrollments for All Courses screen.

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1004238&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by inst-danger** [comment]
#### Release Note: 
Only show active courses in Submit Assignment flow

#### Affected Apps: Student, Teacher

[MBL-13099](https://instructure.atlassian.net/browse/MBL-13099)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 94.95% | 95.01% | -0.06%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Use Cases/GetCourses.swift [hunk 1]
```diff
@@ -21,14 +21,35 @@ import Foundation
...
+    let perPage: Int
 
-    public init(showFavorites: Bool = false, state: [GetCoursesRequest.State] = GetCoursesRequest.State.defaults, perPage: Int = 10) {
-        request = GetCoursesRequest(state: state, perPage: perPage)
-        scope = showFavorites ?
-            .where(#keyPath(Course.isFavorite), equals: true, orderBy: #keyPath(Course.name)) :
-            .all(orderBy: #keyPath(Course.name), ascending: true, naturally: true)
+    public var scope: Scope {
+        if showFavorites {
+            return Scope.where(#keyPath(Course.isFavorite), equals: true, orderBy: #keyPath(Course.name))
```
Comment (vanwagonet) [inline Core/Core/Use Cases/GetCourses.swift]: does this need `ascending: true, naturally: true` like below?

File: rn/Teacher/src/canvas-api/apis/courses.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ import { paginate, exhaust } from '../utils/pagination'
 import httpClient from '../httpClient'
 import { isTeacher } from '../../modules/app'
 
-export function getCourses (): ApiPromise<Course[]> {
+export function getCourses (enrollmentState = 'active'): ApiPromise<Course[]> {
```
Comment (vanwagonet) [inline rn/Teacher/src/canvas-api/apis/courses.js]: this does not appear to be used.

**Pull request merged**